aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -177,13 +178,9 @@ void initialize(
       GlueClient client,
       LockManager lock,
       FileIO io) {
-    Preconditions.checkArgument(
-        path != null && path.length() > 0,
-        "Cannot initialize GlueCatalog because warehousePath must not be null or empty");
-
     this.catalogName = name;
     this.awsProperties = properties;
-    this.warehousePath = LocationUtil.stripTrailingSlash(path);
+    this.warehousePath = path != null ? LocationUtil.stripTrailingSlash(path) : null;
```

Comment:
If the `path` is empty, it fails in `LocationUtil.stripTrailiingSlash(path)`
```suggestion
    this.warehousePath = (path != null && path.length() > 0) ? LocationUtil.stripTrailingSlash(path) : null;
```

Reply:
good point, updated